app/src/main/java/ch/heigvd/pro/b04/android/Poll/Question/QuestionAdapter.java

```diff
@@ -0,0 +1,108 @@
+package ch.heigvd.pro.b04.android.Poll.Question;
+
+import android.graphics.Color;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import android.widget.Button;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LifecycleOwner;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import ch.heigvd.pro.b04.android.Poll.PollViewModel;
+import ch.heigvd.pro.b04.android.R;
+
+public class QuestionAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>  {
+    private static final int VIEW_TYPE_HEADER = 0;
+    private static final int VIEW_TYPE_QUESTION = 1;
+    private PollViewModel state;
+
+    private List<Question> questions = new LinkedList<>();
+
+    public QuestionAdapter(PollViewModel state, LifecycleOwner lifecycleOwner) {
+        this.state = state;
+
+        state.getQuestions().observe(lifecycleOwner, newQuestions -> {
+            for (Question question : newQuestions) {
+                if (!questions.contains(question))
+                    questions.add(question);
```

Comment:
I'd suggest writing `Question q : newQuestions` which would make `if (!questions.contains(q))` clearer.

I must admit it took me waaaay too long to figure out what was happening ;)